### PR TITLE
gdb: Introduce 'scylla range-tombstones' command

### PR DESCRIPTION
**Pull request by tgrabiec**
Prints and validates range tombstones in a given container.

Currently supported containers:

 - mutation_partition

Example:
```
    (gdb) scylla range-tombstones $mp
    {
      start: ['a', 'b'],
      kind: bound_kind::excl_start,
      end: ['a', 'b'],
      kind: bound_kind::incl_end,
      t: {timestamp = 1672546889091665, deletion_time = {__d = {__r = 1672546889}}}
    }
    {
      start: ['a', 'b'],
      kind: bound_kind::excl_start,
      end: ['a', 'c']
      kind: bound_kind::incl_end,
      t: {timestamp = 1673731764010123, deletion_time = {__d = {__r = 1673731764}}}
    }
```

### CODE REVIEW

**Comment by avikivity** [comment]
We can consider [xmethods](https://sourceware.org/gdb/onlinedocs/gdb/Xmethods-In-Python.html) for such things, so they can work at an expression level rather than as command. Not proposing to change this, just an idea for the future.

**Comment by tgrabiec** [comment]
> We can consider [xmethods](https://sourceware.org/gdb/onlinedocs/gdb/Xmethods-In-Python.html) for such things, so they can work at an expression level rather than as command. Not proposing to change this, just an idea for the future.

Which methods would you replace with xmethods here?

**Comment by avikivity** [comment]
> > We can consider [xmethods](https://sourceware.org/gdb/onlinedocs/gdb/Xmethods-In-Python.html) for such things, so they can work at an expression level rather than as command. Not proposing to change this, just an idea for the future.
> 
> Which methods would you replace with xmethods here?

For anything (X) printable, add X::to_string() as an xmethod (and perhaps also regular method). Then to print an X, write `(gdb) print x.to_string()`.

File: scylla-gdb.py [hunk 1]
```diff
@@ -5342,6 +5342,67 @@ def unqualify(name):
...
+        }
+
+    """
+
+    def __init__(self):
+        gdb.Command.__init__(self, 'scylla range-tombstones', gdb.COMMAND_USER, gdb.COMPLETE_COMMAND)
+
+    """value is an instance of clustering_key"""
+    def parse_clustering(self, value):
+        return get_current_schema().parse_clustering_key(value)
```
Comment (denesb) [inline scylla-gdb.py:5376]: If no schema is set this should print a warning and tell the user to use `scylla set_schema` to set it first.

**Review by denesb — APPROVED** [review]
LGTM, just one usability comment.

**Comment by denesb** [comment]
> We can consider [xmethods](https://sourceware.org/gdb/onlinedocs/gdb/Xmethods-In-Python.html) for such things, so they can work at an expression level rather than as command. Not proposing to change this, just an idea for the future.

xmethods would be a good fit for something like `$collection_element()` which is really just an `operator[]()` or `$dereference_smart_ptr()` which is basically `operator*()`. Although convenience functions can also be used in expressions just like xmethods.

File: scylla-gdb.py (hunk 1 above)
Comment (tgrabiec) [reply scylla-gdb.py:5376]: It tells him that in the exception message:

```
def get_current_schema():
    if not current_schema:
        raise Exception('Current schema not set. Use \"scylla set-schema\" to set it.')
    return current_schema

```

**Pull request closed without merging**